Sparse CSR CUDA: Add block torch.addmv when mat is sparse

IvanYashchuk opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #68711
* #68710
* #68709
* __->__ #68708

This PR adds block CSR matrix times dense vector multiplication.

cc @nikitaved @pearu @cpuhrsch @IvanYashchuk @ngimel

Differential Revision: [D32647694](https://our.internmc.facebook.com/intern/diff/D32647694)

IvanYashchuk committed:
Sparse CSR CUDA: Add block torch.addmv when mat is sparse

This PR adds block CSR matrix times dense vector multiplication.

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: Add block torch.addmv when mat is sparse"


This PR adds block CSR matrix times dense vector multiplication.

cc nikitaved pearu cpuhrsch IvanYashchuk

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: Add block torch.addmv when mat is sparse"


This PR adds block CSR matrix times dense vector multiplication.

cc nikitaved pearu cpuhrsch IvanYashchuk ngimel

[ghstack-poisoned]

cpuhrsch commented:
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32647694).

IvanYashchuk commented:
`linux-bionic-cuda10.2` test failures are real.

IvanYashchuk committed:
Update on "Sparse CSR CUDA: Add block torch.addmv when mat is sparse"


This PR adds block CSR matrix times dense vector multiplication.

cc nikitaved pearu cpuhrsch IvanYashchuk ngimel

Differential Revision: [D32647694](https://our.internmc.facebook.com/intern/diff/D32647694)

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: Add block torch.addmv when mat is sparse"


This PR adds block CSR matrix times dense vector multiplication.

cc nikitaved pearu cpuhrsch IvanYashchuk ngimel

Differential Revision: [D32647694](https://our.internmc.facebook.com/intern/diff/D32647694)

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: Add block torch.addmv when mat is sparse"


This PR adds block CSR matrix times dense vector multiplication.

cc nikitaved pearu cpuhrsch IvanYashchuk ngimel

Differential Revision: [D32647694](https://our.internmc.facebook.com/intern/diff/D32647694)

[ghstack-poisoned]

IvanYashchuk committed:
Update on "Sparse CSR CUDA: Add block torch.addmv when mat is sparse"


This PR adds block CSR matrix times dense vector multiplication.

cc nikitaved pearu cpuhrsch IvanYashchuk ngimel

Differential Revision: [D32647694](https://our.internmc.facebook.com/intern/diff/D32647694)

[ghstack-poisoned]

cpuhrsch commented:
@cpuhrsch has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32647694).
